lib/Target/Btor/BtorToBtorIRTranslation.cpp

```diff
@@ -0,0 +1,781 @@
+
+        case BTOR2_TAG_sort: {
+            switch (l->sort.tag) {
+                case BTOR2_TAG_SORT_bitvec:
+                case BTOR2_TAG_SORT_array:
+                default:
+                break;
+            }
+        }
+        break;
+
+        case BTOR2_TAG_state: states.push_back (l); break;
+
+        case BTOR2_TAG_add:
+        case BTOR2_TAG_and:
+        case BTOR2_TAG_concat:
+        case BTOR2_TAG_const:
+        case BTOR2_TAG_constd:
+        case BTOR2_TAG_consth:
+        case BTOR2_TAG_dec:
+        case BTOR2_TAG_eq:
+        case BTOR2_TAG_implies:
+        case BTOR2_TAG_inc:
+        case BTOR2_TAG_ite:
+        case BTOR2_TAG_mul:
+        case BTOR2_TAG_nand:
+        case BTOR2_TAG_neg:
+        case BTOR2_TAG_neq:
+        case BTOR2_TAG_nor:
+        case BTOR2_TAG_not:
+        case BTOR2_TAG_one:
+        case BTOR2_TAG_ones:
+        case BTOR2_TAG_or:
+        case BTOR2_TAG_output:
+        case BTOR2_TAG_redand:
+        case BTOR2_TAG_redor:
+        case BTOR2_TAG_redxor:
+        case BTOR2_TAG_sdiv:
+        case BTOR2_TAG_sext:
+        case BTOR2_TAG_sgt:
+        case BTOR2_TAG_sgte:
+        case BTOR2_TAG_iff:
+        case BTOR2_TAG_slice:
+        case BTOR2_TAG_sll:
+        case BTOR2_TAG_slt:
+        case BTOR2_TAG_slte:
+        case BTOR2_TAG_sra:
+        case BTOR2_TAG_srem:
+        case BTOR2_TAG_srl:
+        case BTOR2_TAG_sub:
+        case BTOR2_TAG_udiv:
+        case BTOR2_TAG_uext:
+        case BTOR2_TAG_ugt:
+        case BTOR2_TAG_ugte:
+        case BTOR2_TAG_ult:
+        case BTOR2_TAG_ulte:
+        case BTOR2_TAG_urem:
+        case BTOR2_TAG_xnor:
+        case BTOR2_TAG_xor:
+        case BTOR2_TAG_zero:
+        case BTOR2_TAG_read:
+        case BTOR2_TAG_write:
+        case BTOR2_TAG_fair:
+        case BTOR2_TAG_justice:
+        case BTOR2_TAG_rol:
+        case BTOR2_TAG_ror:
+        case BTOR2_TAG_saddo:
+        case BTOR2_TAG_sdivo:
+        case BTOR2_TAG_smod:
+        case BTOR2_TAG_smulo:
+        case BTOR2_TAG_ssubo:
+        case BTOR2_TAG_uaddo:
+        case BTOR2_TAG_umulo:
+        case BTOR2_TAG_usubo:
+        default:
+        break;
+    }
+}
+
+static void parse_model () {
+    assert (model_file);
+    model = btor2parser_new ();
+    if (!btor2parser_read_lines (model, model_file))
+    std::cerr << "parse error at: " << btor2parser_error (model) << "\n";
+    num_format_lines = btor2parser_max_id (model);
+    inits.resize (num_format_lines, nullptr);
+    nexts.resize (num_format_lines, nullptr);
+    Btor2LineIterator it = btor2parser_iter_init (model);
+    Btor2Line *line;
+    while ((line = btor2parser_iter_next (&it))) parse_model_line (line);
+
+    for (size_t i = 0; i < states.size (); i++)
+    {
+    Btor2Line *state = states[i];
+    if (!nexts[state->id])
+    {
+        std::cerr << "state " << state->id << " without next function\n";
+    }
+    }
+}
+
+void filterInits() {
```

Comment:
why would there be empty inits? I think this means uninitialized.

This changes order of inits which is probably not correct behavior since `inits` array maps latch id to initial value.

Reply:
from programming perspective, the filtering could have easily be done in place. But I think it is not right to begin with :)

Reply:
The reason for filtering is that when we parse using the method in [btorsim](https://github.com/Boolector/btor2tools/blob/master/src/btorsim/btorsim.cpp#L316), we see that the vectors that hold `inits` and `nexts` are set to a specific size, initialized with `nullptr`. For this reason, it is advantageous to filter out the `nullptrs` while maintaining relative order. 

Can do filtering in place.